src/controllers/krate/search.rs

```diff
@@ -55,28 +55,29 @@ pub async fn search(app: AppState, req: Parts) -> AppResult<Json<Value>> {
         use seek::*;
 
         let params = req.query();
-        let option_param = |s| params.get(s).map(|v| v.as_str());
-        let sort = option_param("sort");
-        let include_yanked = option_param("include_yanked")
+        let option_param = |s| match params.get(s).map(|v| v.as_str()) {
+            Some(v) if v.contains('\0') => Err(bad_request(format!(
+                "parameter {s} cannot contain a null byte"
+            ))),
+            Some(v) => Ok(Some(v)),
+            None => Ok(None),
+        };
```

Comment:
Is it okay to simply write it as:
```suggestion
        let option_param = |s| match params.get(s).map(|v| v.as_str()) {
            Some(v) if v.contains('\0') => Err(bad_request(format!(
                "parameter {s} cannot contain a null byte"
            ))),
            v @ _ => Ok(v), // or just v => Ok(v)
        };
```
Or will it be more difficult to read in this way?